Support enabling teacher notebook when student notebook is disabled

breity opened the pull request:
## Changes
The teacher notebook can now be enabled for a run if the student notebook is disabled.

Closes #155.

## Test
- Disabled student notebook and enable teacher notebook for a run unit in the authoring tool.
- Switch to Teacher Tools and verify that teacher notebook report is visible and functions.
- Try out all the different combinations for turning on and off the teacher and student notebooks and make sure the settings work properly.

breity committed:
Support enabling teacher notebook when student notebook is disabled.
#155

hirokiterashima commented:
I wonder if the original code should have been
```
if (this.NotebookService.isNotebookEnabled('teacherNotebook')) {
```
The null check in that function may be a backwards-compatible check, for projects that did not have a teacherNotebook block in the json?

And ```this.reportEnabled``` should really be looking at ```teacherNotebook.itemTypes.report.enabled``` instead of ```this.notebookConfig.enabled```. 🤦

breity committed:
Updated check for notebook report enabled in Classroom Monitor.
Added back null check to support legacy units without notebook settings.
Check that both teacher notebook and teacher notebook report item type are enabled.
#155

breity commented:
@hirokiterashima Yep, thanks for catching that. The null check in `this.NotebookService.isNotebookEnabled('teacherNotebook')` should be included. I added that back.

Looks like `this.reportEnabled` actually needs both `this.notebookConfig.enabled` and `this.notebookConfig.itemTypes.report.enabled` to be true. Because the unit could have the report enabled but the teacher notebook disabled.

I updated with both of these changes.

hirokiterashima reviewed (approved):
Looks good.
